pkg/skaffold/deploy/resource/logfile.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/logfile"
+)
+
+// withLogFile returns a multiwriter that writes both to a file and a buffer, with the buffer being written to the provided output buffer in case of error
+func withLogFile(container string, out io.Writer, l []string, muted bool) (io.Writer, func([]string), error) {
+	if !muted || len(l) < 4 {
```

Comment:
even though there's only a few lines, should we still write these out to a file anyway?

also, this line should at least have a comment, and maybe move the `4` to a constant. looking at it without having context it looks very out of place

Reply:
umm, i chose not to write to file if lines are less than the number.
re constant: Will move it.

Reply:
ok fair enough i don't feel strongly about it.